src/o365/graph/commands/subscription/subscription-add.ts

```diff
@@ -0,0 +1,254 @@
+        cmd.log(`Actual expiration date time: ${actualExpirationIsoString}`);
+      }
+
+      if (this.verbose) {
+        cmd.log(`An expiration maximum delay is resolved for the resource '${args.options.resource}' : ${resolvedExpirationDelay} minutes.`);
+      }
+
+      return actualExpirationIsoString;
+    }
+
+    // If an resource specific expiration has not been found, return a default expiration delay
+    if (this.verbose) {
+      cmd.log(`An expiration maximum delay couldn't be resolved for the resource '${args.options.resource}'. Will use generic default value: ${DEFAULT_EXPIRATION_DELAY_IN_MINUTES} minutes.`);
+    }
+    const actualExpiration = fromNow(DEFAULT_EXPIRATION_DELAY_IN_MINUTES - SAFE_MINUTES_DELTA);
+    const actualExpirationIsoString = actualExpiration.toISOString();
+    if (this.debug) {
+      cmd.log(`Actual expiration date time: ${actualExpirationIsoString}`);
+    }
+    return actualExpirationIsoString;
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-r, --resource <resource>',
+        description: `The resource that will be monitored for changes`
+      },
+      {
+        option: '-u, --notificationUrl <notificationUrl>',
+        description: 'The URL of the endpoint that will receive the notifications. This URL must use the HTTPS protocol'
+      },
+      {
+        option: '-c, --changeType <changeType>',
+        description: `The type of change in the subscribed resource that will raise a notification. The supported values are: created, updated, deleted. Multiple values can be combined using a comma-separated list`,
+        autocomplete: ['created', 'updated', 'deleted']
+      },
+      {
+        option: '-e, --expirationDateTime [expirationDateTime]',
+        description: `The date and time when the webhook subscription expires. The time is in UTC, and can be an amount of time from subscription creation that varies for the resource subscribed to. If not specified, the maximum allowed expiration for the specified resource will be used`
+      },
+      {
+        option: '-s, --clientState [clientState]',
+        description: `The value of the clientState property sent by the service in each notification. The maximum length is 128 characters`
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(): CommandValidate {
+    return (args: CommandArgs): boolean | string => {
+      if (!args.options.resource) {
+        return 'Required option resource is missing';
+      }
+
+      if (!args.options.notificationUrl) {
+        return 'Required option notificationUrl is missing';
+      }
+
+      if (args.options.notificationUrl.indexOf('https://') != 0) {
+        return `The specified notification URL '${args.options.notificationUrl}' is not a valid Webhook endpoint URL`;
```

Comment:
It would be clearer to say that the notification URL must start with `https://` to make it easier for the user to correct the error. Otherwise it's hard for them to guess what constitutes a valid URL